pkg/storage/bloom/v1/builder.go

```diff
@@ -64,42 +68,105 @@ type SeriesWithBloom struct {
 	Bloom  *Bloom
 }
 
-func (b *BlockBuilder) BuildFrom(itr Iterator[SeriesWithBloom]) (uint32, error) {
-	for itr.Next() {
-		if err := b.AddSeries(itr.At()); err != nil {
-			return 0, err
+type updatableBounds SeriesBounds
+
+func newUpdatableBounds() updatableBounds {
+	return updatableBounds{
+		FromTs:    model.Latest,
+		ThroughTs: model.Earliest,
+		FromFp:    model.Fingerprint(math.MaxInt64),
+		ThroughFp: model.Fingerprint(0),
+	}
+}
+
+func (b *updatableBounds) update(series *Series) {
```

Comment:
Can we add the function to `SeriesBounds` instead making a custom type?

Reply:
That was my first idea but I think it may lead to misusage/bugs. Moving the Update method to `SeriesBounds` would require us to make `newSeriesBounds` to return the `SeriesBounds` initialized as:
https://github.com/grafana/loki/blob/373c9fd7d1a0adcdad706b6362c8d882717d024a/pkg/storage/bloom/v1/builder.go#L73-L80

I think it's better to have this in a specialized type for the matter.